modules/core/src/main/java/org/apache/ignite/maintenance/MaintenanceRegistry.java

```diff
@@ -99,6 +100,27 @@
      */
     public @Nullable MaintenanceTask registerMaintenanceTask(MaintenanceTask task) throws IgniteCheckedException;
 
+    /**
+     * Method to register {@link MaintenanceTask} locally on the node where method is called. If an old task
+     * with the same name exists, applies remapping function to compute a new task.
+     * Has the same restrictions as the {@link #registerMaintenanceTask(MaintenanceTask)}.
+     *
+     *
+     * @param task {@link MaintenanceTask} object with maintenance information that needs
+     *                                     to be stored to maintenance registry.
+     *
+     * @param remappingFunction Function to compute a task if an old {@link MaintenanceTask} with the same name exists.
+     *
+     * @throws IgniteCheckedException If handling or storing maintenance task failed.
+     *
+     * @return Previously registered {@link MaintenanceTask} with the same ID
+     * or null if no tasks were registered for this ID.
+     */
+    public @Nullable MaintenanceTask registerMaintenanceTask(
```

Comment:
Maybe improve **MaintenanceRegistry#registerMaintenanceTask(MaintenanceTask)** with the addition of a **remappingFunction** that, if **null**, will not be executed?

Reply:
I don't see how it's better, because we rarely need remapping function